R/create_lca_test_flags.R

```diff
@@ -0,0 +1,47 @@
+#' Create LCA test flags
+#'
+#' @description Create flags for Local Authorities
+#'
+#' @param data the data containing an LCA variable
+#' @param lca_var LCA variable e.g. CA2019, CA2011
+#'
+#' @return a dataframe with flag (1 or 0) for each LCA
+#' @export
+#' @family flag functions
+create_lca_test_flags <- function(data, lca_var) {
+  data <- data %>%
+    dplyr::mutate(
+      Aberdeen_City = dplyr::if_else({{ lca_var }} == "S12000033", 1, 0),
```

Comment:
```suggestion
      Aberdeen_City = dplyr::if_else({{ lca_var }} == "S12000033" | {{ lca_var }} == "01", 1, 0),
```
This function looks great! 

I've had a thought where this may not work/apply to other lca variables. I'm not too sure if this is applicable to any dataset tests but these codes would only apply to the variable `HL1_sending_lca`. In the suggested output LCA codes in the SLFs are coded "01", "02" etc. Perhaps its worth adding in an `OR` statement so this could be picked up? I'm not sure if its worth doing this so we would need to check if any other tests use LCA. If no other tests use LCA flags, then i think the function is perfect the way it is. From the top of my head maybe in some SC tests we use LCA but that may also be a different code? e.g. `sc_sending_lca`?

Reply:
I had a look at the SC tests but they don't do the same thing, they just look for a missing or not missing sending location as far as I can tell. We could update the sc tests to break down by lca, in which case I can change this code to account for other lca code types, or we can leave both as is

Reply:
> I think the SC tests should probably break down by sending LA (can do that next/separately , no need to include it in this PR).
> 
> We can use `%in%` to get more streamlined code: `Aberdeen_City = dplyr::if_else({{ lca_var }} %in% c("S12000033", "01")`
> 
> Also (maybe a wider point) but is the `if_else` actually needed here? If we're just summing the `1/0` then that's the same as summing `TRUE/FALSE` and that can be achieved without the `if_else` which again cuts down some code:
> 
> ```r
> dplyr::mutate(
>       Aberdeen_City = {{ lca_var }} %in% c("S12000033", "01"),
>       ...
>       )
> ```
> 
> Don't forget to add any 'old' codes to the list too: [Boundary changes lookup.xlsx](https://github.com/Public-Health-Scotland/source-linkage-files/files/9208708/Boundary.changes.lookup.xlsx) that just protects us against some datasets using older codes - for the test we're not interested in which code is being used, we're just interested in properly grouping things into LAs so we can see that breakdown.

Just quoting this here! Also I made an issue for the SC test changes (#320)